pandas/indexes/base.py

```diff
@@ -326,6 +326,22 @@ def __new__(cls, data=None, dtype=None, copy=False, name=None,
                         pass
             # other iterable of some kind
             subarr = _asarray_tuplesafe(data, dtype=object)
+
+            d2 = data
+            if not is_list_like(d2):
+                d2 = [data]
+            # Raises if coercion from unsigned to signed with neg data
```

Comment:
these 2 routines can actually live in ``pandas.types.cast``, esentially all of this boils down to calling.

``
assert_safe_casting(subarr, dtype)
``